test/config-luatest/config_include_test.lua

```diff
@@ -0,0 +1,621 @@
+                        instances = {
+                            ['instance-001'] = {
+                                labels = {
+                                    value_2 = 'from included_config.yaml',
+                                    value_3 = 'from included_config.yaml',
+                                },
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    }
+
+    local config_file = treegen.write_file(
+        g.dir, 'config.yaml', yaml.encode(config))
+    treegen.write_file(
+        g.dir, 'included_config.yaml', yaml.encode(included_config))
+
+    local opts = {
+        config_file = config_file,
+        alias = 'instance-001',
+        chdir = g.dir,
+    }
+    g.server = server:new(opts)
+    g.server:start()
+
+    g.server:exec(function()
+        t.assert_equals(
+            require('config'):get('labels'),
+            {
+                value_1 = 'from config.yaml',
+                value_2 = 'from config.yaml',
+                value_3 = nil,
+            }
+        )
+    end)
+
+    config.include = {'included_config.yaml'}
+    treegen.write_file(g.dir, 'config.yaml', yaml.encode(config))
+
+    g.server:exec(function()
+        require('config'):reload()
+        t.assert_equals(
+            require('config'):get('labels'),
+            {
+                value_1 = 'from config.yaml',
+                value_2 = 'from included_config.yaml',
+                value_3 = 'from included_config.yaml',
+            }
+        )
+    end)
+
+    g.server:grep_log(
+        ('skipping already processed config file: ' ..
+        '%s/config.yaml"'):format(g.dir),
+        1024, {filename = g.dir .. '/var/log/instance-001/tarantool.log'}
+    )
+end
+
+g.test_include_erroneous = function()
+    local config = table.deepcopy(common_config)
+    local errorneous_config = ":"
+
+    local config_file = treegen.write_file(
+        g.dir, 'config.yaml', yaml.encode(config))
+    treegen.write_file(g.dir, 'errorneous_config.yaml', errorneous_config)
+
+    local opts = {
+        config_file = config_file,
+        alias = 'instance-001',
+        chdir = g.dir,
+    }
+
+    g.server = server:new(opts)
+    g.server:start()
+
+    config.include = {'errorneous_config.yaml'}
+    treegen.write_file(g.dir, 'config.yaml', yaml.encode(config))
+
+    t.assert_error_msg_matches(
+        ".*Unable to parse config file .* as YAML.*",
+        g.server.exec, g.server, function()
+            require('config'):reload()
+    end)
+end
+
+g.test_conditional_include = function()
+    -- Remove yaml document start and end markers, because we will need to add
+    -- conditional include to the config, which is not a valid yaml.
+    local common_config_string = yaml.encode(common_config):sub(5,-6)
+
+    local config = [[
+conditional:
+- if: tarantool_version >= 0.0.0
+  include: [included_config.yaml]
+]] .. common_config_string
```

Comment:
```lua
local config = table.copy(common_config)
config.conditional = {
    {
        ['if'] = 'tarantool_version >= 0.0.0',
        ['include'] = {'included_config.yaml'},
    },
}
```

For me it looks like it should do the same.

Reply:
Thanks! I forgot I wanted to change this)